### DISCUSSION TITLE
Scaling rtp-forwarder needs to open many ports?

### DISCUSSION THREAD

**Discussion by psmux** [pion/webrtc#1845] [Q&A] [answered]
Hi,
Thank you for your wonderful library.

I am trying your rtp-forwarder example. To use it for many users,does each user need a seperate pair of udp ports opened? Please advise me.

Thanks

**Comment by Sean-Der** [comment, marked as answer]
Hey @marlocarlo 

You could demux the traffic using attributes on the RTP packets themselves. Commonly people use SSRC to denote different 'sources' and then PayloadTypes for audio/video.

RTP Header is defined [here](https://github.com/pion/rtp/blob/master/packet.go#L16)

**Reply by psmux** [reply]
Thanks @Sean-Der . I am trying to visualize what I need to do to accomplish this. 

So, I would need to 
1.)include the SSRC value in the the WebRTC browser client through RTCRtpContributingSource? 
2.) then check the SSRC value inside the rtp-forwarder code on the RTP header and then...
3.) Should I do anything to redirect the packet to the corresponding SDPs? or will that happen automatically?


I have seen your simulcast example but there it takes in a stream from one user on one page and it splits them based on multiple video steams. I have multiple users with multiple sessions who need to stream to the same port. So, is it possible to demux these 1000 streams and map them to the corresponding sdp file?

If this gets too complicated, as an alternative, would it be a bad idea to open 4 ports per stream(audio,video+2 for sstcp)? ie 4000 ports for 1000 users?


Thanks.

**Reply by Sean-Der** [reply]
I think 4000 ports should be fine! If you run into port exhuastion you could listen on another IP in the loopback range you have up until `127.255. 255.255`

**Reply by psmux** [reply]
@Sean-Der ,
I am facing heavy packet loss when I send a 720p or hd stream which results in a glitched video.

Is there any solution for this?

Thanks..

**Reply by Sean-Der** [reply]
Hi @marlocarlo 

Do you know if the loss is between `ffmpeg -> Pion` or `Pion -> Browser`? Does this occur over the loopback/internet etc.. ?

**Comment by psmux** [comment]
Hi @Sean-Der ,

The issue is between pion and the browser across the internet. The browser is not opened inside the server. So it is between pion on the server and browser opening the webpage hosted there on my system.

When i stream with a much smaller resolution video like 260p,theres no problem but when i increase the resolution to 720p,theres packet loss for ffmpeg. It shows errors that rtp packets are missed and the streamed video is full of artifacts and pixels.

Thanks.
